libs/styles/styles.css

```diff
@@ -284,166 +742,282 @@
-.detail-m,
-.detail-s {
-  font-weight: 700;
-  text-transform: uppercase;
-}
-
-.con-button {
-  background-color: transparent;
-  border-radius: 16px;
-  border: 2px solid var(--text-color);
-  color: var(--text-color);
+  height: var(--m-button-height);
+  box-sizing: border-box; 
   display: inline-block;
-  font-size: 15px;
   font-style: normal;
-  font-weight: 700;
-  line-height: 16px;
-  padding: 5px 14px;
   text-decoration: none;
+  background-color: transparent;
+  border-radius: var(--s2-corner-radius-2000);
+  border: 2px solid var(--s2-gray-800);
+  color: var(--s2-transparent-black-800);
+  font-size: var(--s2-font-size-100);
+  padding: calc((var(--m-button-height) - var(--s2-font-size-100)) / 2) 14px;
+  font-weight: var(--s2-bold-font-weight);
+  min-width: 72px;
 }
 
-.con-button:hover {
-  background-color: var(--color-black);
-  border-color: var(--color-black);
-  color: var(--color-white);
+.con-button.button-justified {
+  display: block;
+  text-align: center;
+  width: 100%;
+}
+
+.con-button:hover, .con-button:active, .con-button:focus {
   text-decoration: none;
+  background-color: var(--s2-gray-100);
+  border-color: var(--s2-gray-900);
+  color: var(--s2-transparent-black-900);
+}
+
+.con-button:focus {
+  outline: 2px solid var(--s2-blue-800);
+}
+
+.con-button[aria-disabled="true"] {
+  background-color: transparent;
+  border-color: var(--color-gray-300);
+  color: var(--s2-transparent-black-400);
 }
 
 .l-button .con-button,
 .con-button.button-l {
-  border-radius: 20px;
-  font-size: 17px;
-  line-height: 20px;
-  min-height: 21px;
-  padding: 7px 18px 8px;
+  --l-button-height: 40px;
+
+  height: var(--l-button-height);
+  font-size: var(--s2-font-size-200);
+  padding: calc((var(--l-button-height) - var(--s2-font-size-200)) / 2) 16px;
+  min-width: 90px;
 }
 
 .xl-button .con-button,
 .con-button.button-xl {
-  border-radius: 25px;
-  font-size: 19px;
-  line-height: 24px;
-  min-height: 28px;
-  padding: 10px 24px 8px;
-}
+  --xl-button-height: 48px;
 
-.con-button.button-justified {
-  display: block;
-  text-align: center;
-  width: 100%;
+  font-size: var(--s2-font-size-300);
+  height: var(--xl-button-height);
+  padding: calc((var(--xl-button-height) - var(--s2-font-size-300)) / 2) 18px;
+  min-width: 108px;
 }
 
-.xxl-button .con-button {
-  border-radius: 30px;
-  font-size: 22px;
-  line-height: 27px;
-  min-height: 27px;
-  padding: 14px 30px 15px;
-}
+.xxl-button .con-button, 
+.con-button.button-xxl { 
+  --xxl-button-height: 56;
 
-.dark .con-button {
-  border-color: var(--color-white);
-  color: var(--color-white);
+  font-size: var(--s2-font-size-400);
+  height: var(--xxl-button-height);
+  padding: calc((var(--xxl-button-height) - var(--s2-font-size-400)) / 2) 20px;
+  min-width: 126px;
 }
 
-.light .con-button {
-  border-color: var(--text-color);
-  color: var(--text-color);
-}
+.xxl-button-mobile .con-button, /* S2 only */
+.con-button.button-xxl { 
+  --xxl-button-height: 56;
```

Comment:
Same missing unit question. And looking at it, they (L827 & L837) seem to share values, could we consolidate them?

Reply:
Good catch! This was actually an incorrect value here.